configuration/cdap-site.xml

```diff
@@ -777,168 +674,292 @@
-      connection when retrieving job status and history
+      Determines the start-up of the CDAP Explore Service (ad-hoc SQL
+      queries); if false, the explore service starts up when CDAP is
+      started; if true, the CDAP Explore Service will start upon the first query
+      it receives
     </description>
+    <value-attributes>
+      <type>boolean</type>
+      <overridable>false</overridable>
+    </value-attributes>
   </property>
 
   <property>
-    <name>mapreduce.include.custom.format.classes</name>
+    <name>explore.writes.enabled</name>
     <value>true</value>
     <description>
-      Indicates whether to include custom input/output format classes in the job.jar or
-      not; if set to true, custom format classes will be added to the job.jar and
-      available as part of the MapReduce system classpath
+      Determines if writing to a table through the CDAP Explore Service (ad-hoc SQL queries) is enabled
     </description>
+    <value-attributes>
+      <type>boolean</type>
+      <overridable>false</overridable>
+    </value-attributes>
   </property>
 
-  <!-- Scheduler related changes -->
-
-  <property>
-    <name>scheduler.max.thread.pool.size</name>
-    <value>100</value>
-    <description>Size of the scheduler thread pool</description>
-  </property>
-
-  <!-- Router configuration -->
+  <!-- Gateway Configuration -->
 
   <property>
-    <name>router.bind.address</name>
-    <value>0.0.0.0</value>
+    <name>app.boss.threads</name>
+    <value>1</value>
     <description>
-      CDAP Router service bind address
+      Number of Netty service boss threads
     </description>
+    <value-attributes>
+      <type>int</type>
+      <minimum>1</minimum>
+      <overridable>false</overridable>
+    </value-attributes>
   </property>
 
   <property>
-    <name>router.bind.port</name>
-    <value>11015</value>
+    <name>app.connection.backlog</name>
+    <value>20000</value>
     <description>
-      CDAP Router service bind port
+      Max connection backlog of CDAP Master
     </description>
   </property>
 
   <property>
-    <name>router.client.boss.threads</name>
-    <value>1</value>
-    <description>The number of boss threads in the CDAP Router service client</description>
+    <name>app.exec.threads</name>
+    <value>20</value>
+    <description>
+      Number of Netty service executor threads
+    </description>
   </property>
 
   <property>
-    <name>router.client.worker.threads</name>
+    <name>app.worker.threads</name>
     <value>10</value>
-    <description>The number of worker threads in the CDAP Router service client</description>
+    <description>
+      Number of Netty service worker threads
+    </description>
   </property>
 
-  <property>
-    <name>router.connection.backlog</name>
-    <value>20000</value>
-    <description>The connection backlog in the CDAP Router service</description>
-  </property>
```

Comment:
All the Kafka config is at the bottom... Mainly because I plan on removing all of it in #83 in favor of the Ambari-supplied Kafka... I'll have to rebase that branch once this one goes to develop.